wdl-engine/src/value.rs

```diff
@@ -0,0 +1,1184 @@
+        assert_eq!(
+            Value::from(12345).coerce(&mut engine, &types, PrimitiveTypeKind::Integer.into()),
+            Some(Value::from(12345))
+        );
+        // Int -> Float
+        assert_eq!(
+            Value::from(12345).coerce(&mut engine, &types, PrimitiveTypeKind::Float.into()),
+            Some(Value::from(12345.0))
+        );
+        // Int -> Boolean (invalid)
+        assert_eq!(
+            Value::from(12345).coerce(&mut engine, &types, PrimitiveTypeKind::Boolean.into()),
+            None
+        );
+    }
+
+    #[test]
+    fn integer_display() {
+        let engine = Engine::default();
+        let types = Types::default();
+
+        assert_eq!(
+            Value::from(12345).display(&engine, &types).to_string(),
+            "12345"
+        );
+        assert_eq!(
+            Value::from(-12345).display(&engine, &types).to_string(),
+            "-12345"
+        );
+    }
+
+    #[test]
+    fn float_coercion() {
+        let mut engine = Engine::default();
+        let types = Types::default();
+
+        // Float -> Float
+        assert_eq!(
+            Value::from(12345.0).coerce(&mut engine, &types, PrimitiveTypeKind::Float.into()),
+            Some(Value::from(12345.0))
+        );
+        // Float -> Int (invalid)
+        assert_eq!(
+            Value::from(12345.0).coerce(&mut engine, &types, PrimitiveTypeKind::Integer.into()),
+            None
+        );
+    }
+
+    #[test]
+    fn float_display() {
+        let engine = Engine::default();
+        let types = Types::default();
+
+        assert_eq!(
+            Value::from(12345.12345)
+                .display(&engine, &types)
+                .to_string(),
+            "12345.12345"
+        );
+        assert_eq!(
+            Value::from(-12345.12345)
+                .display(&engine, &types)
+                .to_string(),
+            "-12345.12345"
+        );
+    }
+
+    #[test]
+    fn string_coercion() {
+        let mut engine = Engine::default();
+        let types = Types::default();
+
+        let value = engine.new_string("foo");
+        let sym = match value {
+            Value::String(sym) => sym,
+            _ => panic!("expected a string value"),
+        };
+
+        // String -> String
+        assert_eq!(
+            value.coerce(&mut engine, &types, PrimitiveTypeKind::String.into()),
+            Some(value)
+        );
+        // String -> File
+        assert_eq!(
+            value.coerce(&mut engine, &types, PrimitiveTypeKind::File.into()),
+            Some(Value::File(sym))
+        );
+        // String -> Directory
+        assert_eq!(
+            value.coerce(&mut engine, &types, PrimitiveTypeKind::Directory.into()),
+            Some(Value::Directory(sym))
+        );
+        // String -> Boolean (invalid)
+        assert_eq!(
+            value.coerce(&mut engine, &types, PrimitiveTypeKind::Boolean.into()),
+            None
+        );
+    }
+
+    #[test]
+    fn string_display() {
```

Comment:
Is there a way to display as a single quoted string? Don't exactly think there should be. I'd be fine only displaying as double-quoted.

Reply:
There isn't; displaying a value is mostly for debugging/development purposes as generally users will see the values coming as the outputs of a workflow or task and then probably write them out as JSON rather than display them.

Reply:
That makes sense!